arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/EagerEffect.kt

```diff
@@ -115,6 +115,11 @@ public interface EagerEffect<out R, out A> {
       toEither().fold({ r -> f(r).bind() }, ::identity)
     }
 
+  public fun <R2> mapLeft(f: (R) -> R2): EagerEffect<R2, A> =
+    eagerEffect {
+      toEither().fold({ shift(f(it)) }, { it })
```

Comment:
can we use here ::identity

Reply:
ok, I don't like that cause what does a reference to an inlined function really even "references" / "points to"? It is just much harder to grasp the meaning of the code / what the compiler will do with those. And I still suspect the `::identity` that existed on `bind()` may had been one of the triggers for #2760 but ok, it is changed now.

I will also keep trying to get an example for the coroutines internal errors that are still happening on shift (even after #2765) but I'm not having much luck with it. It seems to be related to using the fold variant with `error` handler at some depth of effect nesting, but I can't figure out a minimal example. Anyway, that would be part of a different issue / PR if I manage to get a simple reproducible scenario for it.